src/promptflow-core/promptflow/executor/_script_executor.py

```diff
@@ -58,6 +59,9 @@ def __init__(
         logger.debug(f"Init params for script executor: {init_kwargs}")
 
         self._flow_file = flow_file
+        if connections and isinstance(connections, dict):
+            connections = DictConnectionProvider(connections)
+        self._connections = connections
```

Comment:
Since it is actually a provider, can we use `self._provider` instead of `self._connections`?

Reply:
Renamed.